sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala

```diff
@@ -123,7 +123,11 @@ object StringUtils extends Logging {
           val stringToAppend = if (available >= sLen) s else s.substring(0, available)
           strings.append(stringToAppend)
         }
-        length += sLen
+
+        // Cap the length at  ByteArrayMethods.MAX_ROUNDED_ARRAY_LENGTH. Otherwise, we
+        // will overflow length causing StringIndexOutOfBoundsException in the substring call
+        // above.
```

Comment:
@dilipbiswal Sorry and my comment looked ambiguous. I just wanted to leave a note about your comment above;
>  I think, the original intent of keeping this length outside the if block (which i missed and @MaxGekk pointed out) is to keep the true length of the input. So when we are producing a SQL plan, we want to tell users how much was that the size of the explain string and how much we truncated based on max length specification. So caller could call append method even after we have gone past the max.

I couldn't tell why the count-up is placed outside `the if (!atLimit) block`.

Reply:
@maropu Ah .. so you wanted to explain the above comment better in the PR and not in the code ? Right ?

Reply:
The comment in the code side looks okay and how about this?
```
        // Keeps the total length of appended strings. Note that we need to cap the length at
        // `ByteArrayMethods.MAX_ROUNDED_ARRAY_LENGTH`; otherwise, we will overflow
        // length causing StringIndexOutOfBoundsException in the substring call above.
        length = Math.min(length.toLong + sLen, ByteArrayMethods.MAX_ROUNDED_ARRAY_LENGTH).toInt
```